src/include/hermes.h

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ *
+ * Hermes eBPF-based PCIe Accelerator
+ * Copyright(c) 2020 Eideticom, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * The full GNU General Public License is included in this distribution in
+ * the file called "LICENSE".
+ *
+ * Martin Oliveira <martin.oliveira@eideticom.com>
+ *
+ ******************************************************************************/
+
+#ifndef HERMES_H
+#define HERMES_H
+
+#include <stddef.h>
+
+struct hermes_ioctl_argp {
+	size_t len;
+	void *prog;
```

Comment:
Userspace interfaces should always use fixed size types (__u64, __u32, etc). Even for pointers (they'll need to be cast in the code).

Also a good idea to add a flags field... Make sure to check that it is zero and return EINVAL if it is not.